option.go

```diff
@@ -126,3 +129,23 @@ func (o Option[T]) FlatMap(mapper func(value T) Option[T]) Option[T] {
 
 	return None[T]()
 }
+
+// FlatMap executes the mapper function if value is present or returns None if absent.
+func (o Option[T]) MarshalJSON() ([]byte, error) {
+	if o.isPresent {
+		return json.Marshal(o.value)
+	}
+	return json.Marshal(nil)
+}
+
+// FlatMap executes the mapper function if value is present or returns None if absent.
+func (o *Option[T]) UnmarshalJSON(b []byte) error {
```

Comment:
We should differentiate `null` and `""`.

Please use the following function instead:

```go
func (o *Option[T]) UnmarshalJSON(b []byte) error {
	if bytes.Equal(b, []byte("null")) {
		o.isPresent = false
		return nil
	}

	err := json.Unmarshal(b, &o.value)
	if err != nil {
		return err
	}

	o.isPresent = true
	return nil
}
```

Since we don't compare the o.value to zero value of T, we don't need to require T to be a comparable ;)